src/components/UI/dropdown/index.js

```diff
@@ -0,0 +1,84 @@
+import React, { useEffect, useState, useRef } from 'react';
+import Link from 'next/link';
+import styles from './dropdown.module.scss';
+
+const Dropdown = ({
+  isLoggedIn,
+  USER_PROFILE_URL,
+  SIGN_OUT_URL,
+  userData,
+  DEFAULT_AVATAR,
+}) => {
+  const [isOpen, setIsOpen] = useState(false);
+  const toggleDropdown = () => setIsOpen(!isOpen);
```

Comment:
```javascript

const toggleDropdown = () => setIsOpen((prevState) => !prevState);
```

NIT

It's important to perform the state update based on the previous state since react does a batch of state updates.
Very unlikely in our use case, but better to know about it.

Reply:
Thank you for reminding me and I made the changes.